nats-top.yaml

```
@@ -0,0 +1,102 @@
+package:
+  name: nats-top
+  version: "0.6.3"
+  epoch: 0
+  description: A top-like tool for monitoring NATS servers.
+  copyright:
+    - license: MIT
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/nats-io/nats-top
+      tag: v${{package.version}}
+      expected-commit: 812d35cc991232d7606816a15b2ea0b12fb129e7
+
+  - uses: go/build
+    with:
+      packages: .
+      output: nats-top
+      ldflags: -X main.version=${{package.version}} -X main.commit=$(git rev-parse HEAD) -X main.date=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Compat package for ${{package.name}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/local/bin
+          ln -s /usr/bin/nats-top "${{targets.contextdir}}"/usr/local/bin/nats-top
+
+update:
+  enabled: true
+  github:
+    identifier: nats-io/nats-top
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - nats-server
+        - nats
+        - util-linux-misc
+        - mkcert
+        - openssl
+        - python3
+  pipeline:
+    - name: "Version test"
+      runs: |
+        nats-top --version 2>&1 | grep ${{package.version}}
+    - name: Set up nats server and 2 client
+      runs: |
+        nats-server -p 4222 -m 8222 & sleep 2 &
+        nats sub client1 > /dev/null & sleep 2 & nats sub client2 > /dev/null & sleep 2
+    - name: Test and validate nats-top (basic)
+      runs: |
+        nats-top -s localhost -b -u -o nats-output-${{package.version}}.txt
+        grep -q "client1" nats-output-${{package.version}}.txt >/dev/null 2>&1
+        grep -q "client2" nats-output-${{package.version}}.txt >/dev/null 2>&1
+        grep -q "Connections Polled: 2" nats-output-${{package.version}}.txt >/dev/null 2>&1
+    - name: test connection limit (-n)
+      runs: |
+        nats-top -n 1 -o nats-output-${{package.version}}.txt
+        grep -q "Connections Polled: 1" nats-output-${{package.version}}.txt >/dev/null 2>&1
+    - name: test sorting options (-sort)
+      runs: |
+        nats-top -sort uptime -u -o nats-output-${{package.version}}.txt
+        grep -q client2 <(tail -1 nats-output-${{package.version}}.txt)
+    - name: test output with delimiter (-l)
+      runs: |
+        nats-top -o nats-output-${{package.version}}.csv -l ","
+        if [ ! -s "nats-output-${{package.version}}.csv" ]; then
+          echo "Error: CSV File is empty!"
+          exit 1
+        fi
+        # Sanitize the input
+        sed -i '/^$/d' nats-output-${{package.version}}.csv
+        # Ensure at least one column for each rows
+        python -c "import csv; reader = csv.reader(open('nats-output-${{package.version}}.csv', 'r')); [print(row[0]) for row in reader]"
+    - name: test setting refresh delay (-d) and refresh count (-r)
+      runs: |
+        export TERM=xterm && timeout 3 script -q -c "nats-top -d 1 -r 3" /dev/null
```

```suggestion
        export TERM=xterm
        script --quiet --return --command "timeout 3 nats-top -d 1 -r 3"
```
`--return` propagates the exit code from timeout that will propagate the exit code from `nats-top`, so test will fail when expected

---

Thank you for suggesting this! We can use `script` for running any interactive tests now, for future cases.

---

Yes, that's awesome indeed @AnmolVirdi, thanks